### ISSUE TITLE
[`pandas-vet`] PD003/PD004 false positives on xarray objects when pandas is imported

### ISSUE THREAD

**Issue by pums974** [astral-sh/ruff#26264]
### Summary

I believe the rules PD003 and PD004 should be enforced on pandas objects, now it is enforced on all objects as soon as pandas is imported.

for exemple 

```
import pandas as pd
import xarray as xr

arr = xr.DataArray([1.0, float("nan")])

arr.notnull()  # PD004, but arr.notna() does not exist
arr.isnull()   # PD003, but arr.isna() does not exist
```

**Comment by ntBre** [comment]
Yes, this is a very common issue with the `PD` and `NP` rules. Without robust type inference, we don't have a very good way of determining if an object is from pandas or not, and the import is one heuristic to try to make the situation a little better. I think the canonical issue for this general problem was https://github.com/astral-sh/ruff/issues/6432, but I guess we can use this one since that's now closed.

**Comment by pums974** [comment]
Sorry I missed #6432. 

I get the problem now

**Comment by ntBre** [comment]
No worries, thanks for the report!
